### DISCUSSION TITLE
Looking for the Best Method of Modeling Sites and Locations

### DISCUSSION THREAD

**Discussion by adparis99** [netbox-community/netbox#10062] [Q&A] [answered]
Hello,

Our organization has a number of different sites it owns. Some of these sites have a over a dozen different buildings. Our thought process thus far in modeling this in NetBox is to have the entire property be it's own site, with each building serving as a location and network closets serving as sub-locations. This has worked perfectly for us until we began dipping my toes into configuration contexts, which are applicable to sites and not locations. We have two networks at many of our sites, so assigning contexts based on site alone won't always work. I've seen in some feature requests the reasons behind not allowing them to be assigned to racks and understand them, but it seems like it would make sense to assign them locations, unless I am not using the site/location model correctly. 

Just curious on what you all think about how I'm making use of that model and how you use it in your environment. Thanks in advance!

**Comment by jcollie** [comment]
That's the same model I'm using. I believe that I saw in the v3.3 release
notes that you can now assign config contexts to locations, so your problem
may be fixed by upgrading to the latest version.

Jeff

On Thu, Aug 18, 2022 at 2:03 PM adparis99 ***@***.***> wrote:

> Hello,
>
> Our organization has a number of different sites it owns. Some of these
> sites have a over a dozen different buildings. Our thought process thus far
> in modeling this in NetBox is to have the entire property be it's own site,
> with each building serving as a location and network closets serving as
> sub-locations. This has worked perfectly for us until we began dipping my
> toes into configuration contexts, which are applicable to sites and not
> locations. I've seen in some feature requests the reasons behind not
> allowing them to be assigned to racks and understand them, but it seems
> like it would make sense to assign them locations, unless I am not using
> the site/location model correctly.
>
> Just curious on what you all think about how I'm making use of that model
> and how you use it in your environment. Thanks in advance!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/netbox-community/netbox/discussions/10062>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFUVNRYENV2VUWCATIZ7QDVZ2CHVANCNFSM566GWEVQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Jeff Ollie
The majestik møøse is one of the mäni interesting furry animals in Sweden.

**Reply by adparis99** [reply]
Good to know, thank you. Upgrading should be much simpler than reorganizing everything

**Comment by jeremystretch** [comment, marked as answer]
> I believe that I saw in the v3.3 release notes that you can now assign config contexts to locations

Correct; config contexts can be associated with locations in NetBox v3.3 and later.

**Reply by adparis99** [reply]
I see, thank you! I didn't even realize 3.3 was released. Looking forward to checking it out

**Comment by DanSheps** [comment]
So, the way I model sites and location is a little different.  Since sites have distinct addresses, each building is a site normally:

Country: Region 
Province: Region
City: Region
Campus: Region
Building: Site
Floor: Location
Room: Location
Cage: Location
Rack: Rack

You can define a VLAN group for stretched VLANs
